static/js/src/ua-product-selector.js

```diff
@@ -0,0 +1,492 @@
+import { StateManager } from "./utils/state.js";
+import { debounce } from "./utils/debounce.js";
+
+function productSelector() {
+  const form = document.querySelector(".js-shop-form");
+  const stepClassPrefix = ".js-shop-step--";
+  const products = window.productList;
+
+  const addStep = form.querySelector(`${stepClassPrefix}add`);
+  const cartStep = document.querySelector(`${stepClassPrefix}cart`);
+  const formHeader = form.querySelector(".js-shop--form-heading");
+  const shopHeroElement = document.querySelector(".js-shop-hero");
+  const publicCloudElements = form.querySelectorAll(".js-public-cloud-info");
+  const quantityTypeEl = form.querySelector(".js-type-name");
+  const steps = ["type", "quantity", "version", "support", "add", "cart"];
+
+  let state = new StateManager(steps, render);
+
+  init();
+  render();
+
+  function init() {
+    const productInputs = form.querySelectorAll(".js-product-input");
+    const versionTabs = form.querySelectorAll(
+      ".js-shop-step--version .p-tabs__link"
+    );
+
+    productInputs.forEach((input) => {
+      input.addEventListener("input", (e) => {
+        if (input.getAttribute("type") === "radio") {
+          setSelectedRadioClass(input);
+        }
+
+        handleStepSpecificAction(e.target);
+      });
+    });
+
+    document.addEventListener("input", (e) => {
+      if (e.target && e.target.classList.contains("js-product-input")) {
+        handleStepSpecificAction(e.target);
+      }
+    });
+
+    versionTabs.forEach((tab) => {
+      tab.addEventListener("click", (e) => {
+        e.preventDefault();
+        state.set("version", [tab.getAttribute("href")]);
+      });
+    });
+
+    document.addEventListener("click", (e) => {
+      if (e.target && e.target.classList.contains("js-cart-action")) {
+        e.preventDefault();
+        handleCartAction(e.target.dataset);
+      }
+    });
+  }
+
+  function buildCartHTML(lineItems) {
+    const subtotal = calculateSubtotal(lineItems);
+    let lineItemsHTML = "";
+
+    lineItems.forEach((lineItem) => {
+      lineItemsHTML += `<li class="p-list__item">${buildLineItemHTML(
+        lineItem.get("productId")[0],
+        lineItem.get("quantity")[0],
+        lineItem.get("imageURL")[0],
+        "remove"
+      )}</li>`;
+    });
+
+    return `<div class="row">
+      <div class="col-12">
+        <h2>Your subscription so far:</h2>
```

Comment:
Labels end in colons, headings don’t.
```suggestion
        <h2>Your subscription so far</h2>
```